packages/nodes-base/nodes/Code/TaskRunnerSandbox.ts

```diff
@@ -10,28 +10,44 @@ import { isWrappableError, WrappedExecutionError } from './errors/WrappedExecuti
 import { validateNoDisallowedMethodsInRunForEach } from './JsCodeValidator';
 
 /**
- * JS Code execution sandbox that executes the JS code using task runner.
+ * Code execution sandbox that executes the JS code using task runner.
  */
-export class JsTaskRunnerSandbox {
+export class TaskRunnerSandbox {
 	constructor(
-		private readonly jsCode: string,
+		private readonly language: 'javascript' | 'python',
+		private readonly code: string,
 		private readonly nodeMode: CodeExecutionMode,
 		private readonly workflowMode: WorkflowExecuteMode,
 		private readonly executeFunctions: IExecuteFunctions,
 		private readonly chunkSize = 1000,
 	) {}
 
-	async runCodeAllItems(): Promise<INodeExecutionData[]> {
+	/**
+	 * Run a script by forwarding it to a task runner, together with input items.
+	 *
+	 * The Python runner receives input items together with the task, whereas the
+	 * JavaScript runner does _not_ receive input items together with the task and
+	 * instead retrieves them later, only if needed, via an RPC request.
+	 */
+	async runUsingIncomingItems() {
+		return await this.runCodeAllItems({ items: this.executeFunctions.getInputData() });
+	}
```

Comment:
I think this would actually be much clearer if we didn't have this method. Instead pass the items from Code.node in the native Python use case.

Reply:
Sorry, I don't quite understand this suggestion. `runUsingIncomingItems` is indeed passing the incoming items to the Python runner via `runCodeAllItems` in the native Python case.

Reply:
I meant remove this method. In Code.node.ts, instead of:

```ts
return [await sandbox.runUsingIncomingItems()];
```

do:
```ts
return [await sandbox.runCodeAllItems({ items: this.getInputData() })];
```

Having 3 separate methods in the sandbox class for these different things makes the interface confusing

Reply:
My reasoning for adding a new umbrella method is that `runCodeAllItems` is specific to all-items mode, but for the native Python case we send all items for both modes, all-items and per-item.

Reply:
Will get on it soon 👍🏻